### PR TITLE
Improve Python selftests

### PR DESCRIPTION
**Pull request by teknico**
1) Allow running tests in just one subdirectory;
2) move exit code checks to one place;
3) add a TestCase superclass for tests needing a long-running process;
4) remove usage of the `sh` library; it is flaky when handling long-running processes;
5) improve test diagnostics in case of failure.

### CODE REVIEW

File: src/program/lwaftr/tests/subcommands/monitor_test.py [hunk 1]
```diff
@@ -17,37 +16,26 @@
...
+        '--conf', str(DATA_DIR / 'icmp_on_fail.conf'),
         '--on-a-stick', SNABB_PCI0,
         '--mirror', TAP_IFACE,
     )
-
-    monitor_cmd_args = (
-        SNABB_CMD, 'lwaftr', 'monitor',
-        '--name', 'monitor_test',
-        'all',
-    )
```
Comment (dpino) [inline src/program/lwaftr/tests/subcommands/monitor_test.py:35]: Ideally the monitor test needs more test cases: keyword 'none', set by IPv4 address and setting by PID. I'm fine if those test cases are added in a new PR.

File: src/program/lwaftr/tests/subcommands/loadtest_test.py [hunk 2]
```diff
@@ -8,48 +8,40 @@
...
-        BENCHDATA_DIR / 'ipv4_and_ipv6_stick_imix.pcap', 'ALL', 'ALL', SNABB_PCI1,
+        str(BENCHDATA_DIR / 'ipv4_and_ipv6_stick_imix.pcap'), 'ALL', 'ALL',
+        SNABB_PCI1,
     )
-
-    # Use setUpClass to only setup the "run" daemon once for all tests.
-    @classmethod
-    def setUpClass(cls):
-        cls.run_cmd = sh.sudo(*cls.run_cmd_args, _bg=True)
+    wait_for_daemon_startup = True
```
Comment (dpino) [inline src/program/lwaftr/tests/subcommands/loadtest_test.py:39]: It this line meant to be in `setUpClass` body or at class level? If it's a class level I think it's more readable to move it before the method definition, together with other class properties such as `loadtest_args`, `daemon_args`, etc

File: src/program/lwaftr/tests/subcommands/monitor_test.py [hunk 3]
```diff
@@ -17,37 +16,26 @@
...
-        self.assertIn('255.255.255.255', output)
-
-    @classmethod
-    def tearDownClass(cls):
-        cls.run_cmd.terminate()
+        monitor_args = list(self.monitor_args)
+        monitor_args.append(str(self.daemon.pid))
+        output = self.run_cmd(monitor_args)
+        self.assertIn(b'Mirror address set', output,
+            b'\n'.join((b'OUTPUT', output)))
```
Comment (dpino) [inline src/program/lwaftr/tests/subcommands/monitor_test.py:36]: I don't get this part. Is it checking the output matches "Mirror address set"? Wasn't that asserted in `test_monitor` method? If that's the case, what's the reason of checking it twice?

**Review by dpino — APPROVED** [review]
LGTM, although I left some comments about some parts I didn't understand.

File: src/program/lwaftr/tests/subcommands/monitor_test.py (hunk 3 above)
Comment (teknico) [reply src/program/lwaftr/tests/subcommands/monitor_test.py:36]: Not sure what you mean: it *is* asserted in this `test_monitor` method, and only here. I removed the `setupClass` and `tearDownClass` methods from this `TestCase`, because it now relies on `BaseTestCase`.

File: src/program/lwaftr/tests/subcommands/loadtest_test.py (hunk 2 above)
Comment (teknico) [reply src/program/lwaftr/tests/subcommands/loadtest_test.py:39]: It is at class level, and it already is before method definitions, together with the other class properties, as you suggest. Now I'm not sure we're looking at the same changes. :-)
